src/client/pages/OfferNew/utils.ts

```diff
@@ -347,37 +347,33 @@ export const getInsuranceTitle = (
   ]()
 }
 
-export const maskAndFormatRawSsn = (ssn: string) => {
-  if (ssn.length !== 12) {
-    return ssn
-  }
-
-  const CENTURY_LENGTH = 2
-  const DATE_LENGTH = 6
-  return ssn.substr(CENTURY_LENGTH, DATE_LENGTH) + '-****'
+type FormattedBirthdateParams = {
+  birthdate: string
+  currentLocale: string
 }
 
-export const ssnLengthByMarket: Record<Market, number> = {
-  SE: 12,
-  NO: 11,
-  DK: 10,
-}
+export const getFormattedBirthdate = ({
+  birthdate,
+  currentLocale,
+}: FormattedBirthdateParams) => {
+  const localeBirthdateFormat =
+    locales[currentLocale as LocaleLabel].birthdate.formatRegex
 
-export const ssnRegExpByMarket: Record<Market, RegExp> = {
-  SE: /^([1-2][0-9])?[0-9]{2}[0-1][0-9][0-9]{2}[-+]?[0-9]{4}$/,
-  NO: /^[0-9]{2}[0,1][0-9][0-9]{2}[ ]?[0-9]{5}$/,
-  DK: /^[0-9]{2}[0,1][0-9][0-9]{2}[ ]?[0-9]{4}$/,
-}
+  const hasCorrectFormat = localeBirthdateFormat.test(birthdate)
 
-export const createSsnValidator = (market: Market) => (
-  ssn: string,
-): boolean => {
-  switch (market) {
-    case Market.No:
-      return ssnRegExpByMarket[Market.No].test(ssn)
-    case Market.Dk:
-      return ssnRegExpByMarket[Market.Dk].test(ssn)
-    case Market.Se:
-      return ssnRegExpByMarket[Market.Se].test(ssn)
+  if (hasCorrectFormat) {
+    return birthdate
   }
+
+  const defaultFormat = birthdateFormats.default // This is the format we expect from back-end
+  const hasExpectedFormat = defaultFormat.test(birthdate)
+
+  if (!hasExpectedFormat) {
+    throw `Format of birthdate "${birthdate}" doesn't match the expected default format YYYY-MM-DD`
```

Comment:
Are we strict about throwing only `Error`? If we just throw `string` we miss out on nice debugging features (or can Sentry still handle that?)

Reply:
Ah, cool! Didn't know the difference 😳 Not sure if Sentry catches this, will investigate 🕵️‍♀️

Reply:
Sure, I believe it's especially stack traces that are missing unless you `throw new Error(message)`

> The fundamental benefit of Error objects is that they automatically keep track of where they were built and originated.

🙂

Reply:
Aha, nice! Will change that! 🙏